File: week1/eunbeen/가장큰수.cs
using System;
using System.Text;
using System.Linq;

public class Solution {
    public string solution(int[] numbers) {
        var answer = new StringBuilder();
        
        Array.Sort(numbers, compare);
        
        if(numbers[numbers.Length-1] == 0) return "0";
        for(int i=numbers.Length-1; i>=0; i--){
            answer.Append(numbers[i].ToString());
        }
        
        
        return answer.ToString();
    }
    public int compare(int a, int b){
        string ab = a.ToString() + b.ToString();
        string ba = b.ToString() + a.ToString();
        
        return string.Compare(ab, ba);
    }
}

File: week1/eunbeen/가장큰수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        var answer = new StringBuilder();
        List<String> array = new ArrayList();
        
        for(int num : numbers){
            array.add(String.valueOf(num));
        }
        
        Collections.sort(array, (o1, o2) -> {
            String a = o1 + o2;
            String b = o2 + o1;
            
            return b.compareTo(a);
        });
        
        if(array.get(0).equals("0")) return "0";
        for(String num : array){
            answer.append(num);
        }
        
        return answer.toString();
    }
}